zcash_primitives/src/legacy/keys.rs

```diff
@@ -320,6 +321,27 @@ impl NonHardenedChildIndex {
     pub fn index(&self) -> u32 {
         self.0
     }
+
+    pub fn next(&self) -> Option<Self> {
+        Self::from_index(self.0 + 1)
```

Comment:
It would be useful to have a comment here to note that `self.0 + 1` doesn't need to use `checked_add` because this addition can't overflow, and would instead flow into the hardened range, which is then excluded by `Self::from_index`. Just to help casual inspection.